lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.py

```diff
@@ -49,47 +53,158 @@ def createPrimAndAttributes(stage, primPath, mayaReferencePath, mayaNamespace, m
+    # Note: if we are given a group prim to create, then we know that the
+    #       Maya Reference prim name will be unique since it will be the
+    #       only child (of the newly created group prim).
+    checkName = mayaUsd.ufe.uniqueChildName(parentPrim, mayaReferencePrimName) if groupPrim is None else mayaReferencePrimName
     if checkName != mayaReferencePrimName:
-        cmds.error('Maya Reference prim name already exists.')
-        return
+        errorMsgFormat = getMayaUsdLibString('kErrorMayaRefPrimExists')
+        errorMsg = cmds.format(errorMsgFormat, stringArg=(mayaReferencePrimName, ufePathStr))
+        om.MGlobal.displayError(errorMsg)
+        return Usd.Prim()
     validatedPrimName = Tf.MakeValidIdentifier(checkName)
 
     # Extract the USD path segment from the UFE path and append the Maya
     # reference prim to it.
-    ufePath = ufe.PathString.path(ufePathStr)
     parentPath = str(ufePath.segments[1]) if ufePath.nbSegments() > 1 else ''
-    primPath = Sdf.AssetPath(parentPath + '/' + validatedPrimName)
 
     stage = mayaUsd.ufe.getStage(ufePathStr)
 
-    variantSetName = Tf.MakeValidIdentifier(variantSetName)
-    variantName = Tf.MakeValidIdentifier(variantName)
+    # Optionally insert a Group prim as a parent of the Maya reference prim.
+    groupPrim = None
+    if groupPrimName:
+        groupPath = Sdf.AssetPath(parentPath + '/' + groupPrimName)
+        try:
+            groupPrim = stage.DefinePrim(groupPath.path, groupPrimType)
+        except (Tf.ErrorException):
+            groupPrim = Usd.Prim()
+        if not groupPrim.IsValid():
+            errorMsgFormat = getMayaUsdLibString('kErrorCreatingGroupPrim')
+            errorMsg = cmds.format(errorMsgFormat, stringArg=(ufePathStr))
+            om.MGlobal.displayError(errorMsg)
+            return Usd.Prim()
+        if groupPrimKind:
+            model = Usd.ModelAPI(groupPrim)
+            model.SetKind(groupPrimKind)
+
+    if groupPrim:
+        primPath = Sdf.AssetPath(groupPrim.GetPath().pathString + '/' + validatedPrimName)
+    else:
+        primPath = Sdf.AssetPath(parentPath + '/' + validatedPrimName)
+
+    # Were we given a Variant Set to create?
+    variantSetName = None
+    variantName = None
+    if variantSet and (len(variantSet) == 2):
+        variantSetName, variantName = variantSet
     if variantSetName and variantName:
-        ufePathPrim = mayaUsd.ufe.ufePathToPrim(ufePathStr)
-        vset = ufePathPrim.GetVariantSet(variantSetName)
-        vset.AddVariant(variantName)
-        vset.SetVariantSelection(variantName)
+        validatedVariantSetName = Tf.MakeValidIdentifier(variantSetName)
+        validatedVariantName = Tf.MakeValidIdentifier(variantName)
+
+        # If we created a group prim add the variant set there, otherwise add it
+        # to the prim that corresponds to the input ufe path.
+        variantPrim = groupPrim if groupPrim else mayaUsd.ufe.ufePathToPrim(ufePathStr)
```

Comment:
Comment says it, but worth pointing out that if we have a group, the variant is added to group (instead of prim you RMB on).